src/app/_services/depublication.service.ts

```diff
@@ -0,0 +1,129 @@
+
+    formData.append('depublicationFile', file);
+
+    return this.http
+      .post(url, formData, {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        },
+        observe: 'events',
+        params: {
+          clientFilename: file.name,
+          mimeType: file.type
+        },
+        reportProgress: true
+      })
+      .pipe(map(this.handleUploadEvents))
+      .pipe(this.errors.handleRetry());
+  }
+
+  /** setPublicationInfo
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {string} toDepublish - depublication record urls
+  */
+  setPublicationInfo(datasetId: string, toDepublish: string): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/depublished_records/${datasetId}`;
+    return this.http
+      .post<boolean>(
+        url,
+        { toDepublish },
+        {
+          headers: {
+            'Content-Type': 'text/plain'
+          }
+        }
+      )
+      .pipe(this.errors.handleRetry());
+  }
+
+  /** parseFilterParameter
+  /*  return a url parameter representation of a filter string or an empty string
+  /*  @param {string} p - optional
+  */
+  parseFilterParameter(p?: string): string {
+    return p ? `&searchQuery=${p}` : '';
+  }
+
+  /** parseSortParameter
+  /*  return a url parameter representation of a SortParameter or an empty string
+  /*  @param {SortParameter} p - optional
+  */
+  parseSortParameter(p?: SortParameter): string {
+    return p ? `&sortField=${p.field}&sortDirection=${p.direction.toLowerCase()}` : '';
+  }
+
+  /** getPublicationInfo
+  /*  retrieve publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {number} page - the pagination target
+  /*  @param {SortParameter} sort - optional - sort parameter
+  /*  @param {string} filter - optional - sort parameter
+  */
+  getPublicationInfo(
+    datasetId: string,
+    page: number,
+    sort?: SortParameter,
+    filter?: string
+  ): Observable<Results<RecordPublicationInfo>> {
+    const sortParam = this.parseSortParameter(sort);
+    const filterParam = this.parseFilterParameter(filter);
+    const url = `${apiSettings.apiHostCore}/depublished_records/${datasetId}?page=${page}${sortParam}${filterParam}`;
+    return this.http.get<Results<RecordPublicationInfo>>(url).pipe(this.errors.handleRetry());
+  }
+
+  /** getPublicationInfoUptoPage
+  /*  paginated record publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {number} endPage - the number of pages to fetch
+  /*  @param {SortParameter} sort - optional - sort parameter
+  /*  @param {string} filter - optional - sort parameter
+  */
+  getPublicationInfoUptoPage(
+    datasetId: string,
+    endPage: number,
+    sort?: SortParameter,
+    filter?: string
+  ): Observable<MoreResults<RecordPublicationInfo>> {
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
I assume that this comment disables that lint warning for the next line? In java we have the habit of putting another comment explaining why we think that a particular exclusion of the code quality rule is warranted here. Would that be a good idea here too?

Reply:
Removed - it's been changed so it isn't needed